src/paragraph/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { PureComponent } from 'react'
+import cc from 'classcat'
+
+import Text, { TextTagType } from 'text'
+import Button from 'button'
+
+import style from 'paragraph/style'
+
+interface ParagraphProps {
+  readonly className?: Classcat.Class
+  readonly children: string
+  readonly expandable?: boolean
+  readonly expandLabel?: string
+}
+
+interface ParagraphState {
+  expanded: boolean
+}
+
+class Paragraph extends PureComponent <ParagraphProps, ParagraphState> {
+  static MAX_CHAR_SIZE: number = 180
```

Comment:
Line detection is a mess in JS, especially to add the ellipsis at the right place. There is no CSS rule that allow us to do that easily (`text-overflow: ellipsis` only works for single-line components, and `line-clamp` rules have an awful browser support). This will require a complex algorithm based on component width, height and text-size =/

Reply:
Actually since you know the line-height, you know the size you are looking for. So by splitting the string using spacing & dichotomy you can find the right one. 
I think it's not really great to add an ellipsis in the middle of the screen like that, it feels weird